File: solutions/Javascript/Validate Binary Search Tree.js
function Node(data) {
  this.left = null;
  this.right = null;
  this.val = data;
}

var isValidBST = (root) => {
  function helper(root, left, right) {
    if (root === null) return true;
    else if (
      (left != null && left < root.val) ||
      (right != null && right > root.val)
    )
      return false;
    else
      return (
        helper(root.left, root.value, right) &&
        helper(root.right, left, root.right)
      );
  }
  return helper(root, null, null);
};

File: solutions/python/Validate Binary Search Tree.py
class Node:
    def __init__(self,data):
        this.left=None
        this.right=None
        this.val=data

def isValidBST(root:Node)->bool:
    return helper(root,None,None)
def helper(root,left,right):
    if root==None:
         return True
    elif left!=None and root.left>left or right!=None and root.right<right:
        return False
    else:
        return helper(root,root.left,right) and helper(root.right,left,root.right)




